autolens/plot/plotters/subhalo_plotters.py

```diff
@@ -0,0 +1,266 @@
+        """
+        number_subplots = 6
+
+        self.open_subplot_figure(number_subplots=number_subplots)
+
+        self.setup_subplot(number_subplots=number_subplots, subplot_index=1)
+
+        self.mat_plot_2d.title.kwargs["label"] = "Normalized Residuals (No Subhalo)"
+
+        fit_imaging_plotter_before = self.fit_imaging_plotter_from(
+            fit_imaging=fit_imaging_before
+        )
+
+        fit_imaging_plotter_before.figure_normalized_residual_map()
+
+        self.setup_subplot(number_subplots=number_subplots, subplot_index=2)
+
+        self.mat_plot_2d.title.kwargs["label"] = "Chi-Squared Map (No Subhalo)"
+
+        fit_imaging_plotter_before.figure_chi_squared_map()
+
+        self.setup_subplot(number_subplots=number_subplots, subplot_index=3)
+
+        self.mat_plot_2d.title.kwargs["label"] = "Source Reconstruction (No Subhalo)"
+
+        fit_imaging_plotter_before.figure_plane_image_of_plane(plane_index=1)
+
+        fit_imaging_plotter_detect = self.fit_imaging_plotter_from(
+            fit_imaging=fit_imaging_detect
+        )
+
+        self.setup_subplot(number_subplots=number_subplots, subplot_index=4)
+
+        self.mat_plot_2d.title.kwargs["label"] = "Normailzed Residuals (With Subhalo)"
+
+        fit_imaging_plotter_detect.figure_normalized_residual_map()
+
+        self.mat_plot_2d.title.kwargs["label"] = "Chi-Squared Map (With Subhalo)"
+
+        self.setup_subplot(number_subplots=number_subplots, subplot_index=5)
+
+        fit_imaging_plotter_detect.figure_chi_squared_map()
+
+        self.mat_plot_2d.title.kwargs["label"] = "Source Reconstruction (With Subhalo)"
+
+        self.setup_subplot(number_subplots=number_subplots, subplot_index=6)
+
+        fit_imaging_plotter_detect.figure_plane_image_of_plane(plane_index=1)
+
+        self.mat_plot_2d.output.subplot_to_figure()
+
+        self.mat_plot_2d.figure.close()
+
+
+def agg_max_log_likelihood_from_aggregator(aggregator):
+
+    samples = list(filter(None, aggregator.values("samples")))
+    log_likelihoods = [max(samps.log_likelihoods) for samps in samples]
+    index = np.argmax(log_likelihoods)
+    search_max = list(filter(None, aggregator.values("search")))[index]
+
+    directory = str(search_max.paths.name)
+    directory = directory.replace(r"/", path.sep)
+
+    return aggregator.filter(aggregator.directory.contains(directory))
+
+
+def copy_pickle_files_to_agg_max(agg_max_log_likelihood):
+
+    search_max_log_likelihood = list(agg_max_log_likelihood.values("search"))
+    pickle_path_max_log_likelihood = search_max_log_likelihood[0].paths.pickle_path
+
+    pickle_path_max_log_likelihood = str(pickle_path_max_log_likelihood).replace(
+        r"/", path.sep
+    )
+
+    pickle_path_grid_search = pickle_path_max_log_likelihood
+
+    pickle_path_grid_search = path.split(pickle_path_grid_search)[0]
+    pickle_path_grid_search = path.split(pickle_path_grid_search)[0]

```

Comment:
lol wtf is this!?

Reply:
When the seqel aggregator is up and running and supports grid searches it will be deleted  lol.